server/docker_resource.py

```diff
@@ -0,0 +1,433 @@
+
+            title='Updating Settings ',
+            type='HistomicsTK.images',
+            user=getCurrentUser(),
+            public=True,
+            async=True
+        )
+        ModelImporter.model('job', 'jobs').scheduleJob(job)
+
+
+def getDockerImageSettings():
+    module_list = ModelImporter.model('setting').get(
+        PluginSettings.DOCKER_IMAGES)
+    if module_list is None:
+        module_list = {}
+    return module_list
+
+
+class DockerCache():
+
+    imageName = 'docker_image_name'
+    type = 'type'
+    xml = 'xml'
+    cli_dict = 'cli_list'
+
+    def __init__(self, cache):
+        """
+        Data is stored in the following format:
+            {image_name_hash:
+                {
+                cli_list:{
+                    cli_name:{
+                        type:<type>
+                        xml:<xml>
+
+                            }
+                        }
+                docker_image_name:<name>
+                }
+            }
+        """
+        if isinstance(cache, dict):
+            self.data = cache
+        elif isinstance(cache, str):
+            if cache != '':
+                self.data = json.loads(cache)
+        else:
+            self.data = {}
+
+    def getDockerImageList(self):
+
+        return [str(imgDict[DockerCache.imageName])
+                for (imgHash, imgDict) in iteritems(self.data)]
+
+    def imageAlreadyLoaded(self, name):
+        imageKey = self._getHashKey(name)
+        if imageKey in self.data:
+            return True
+        else:
+            return False
+
+    def addImage(self, name):
+
+        imageKey = self._getHashKey(name)
+        self.data[imageKey] = {}
+
+        self.data[imageKey][DockerCache.imageName] = name
+        self.data[imageKey][DockerCache.cli_dict] = {}
+
+    def addCLI(self, img_name, cli_name, type, xml):
+
+        cliData = {}
+        cliData[DockerCache.type] = type
+        cliData[DockerCache.xml] = xml
+
+        imageKey = self._getHashKey(img_name)
+        self.data[imageKey][DockerCache.cli_dict][cli_name] = cliData
+
+    def raw(self):
+        return self.data
+
+    def deleteImage(self, name):
+        imageKey = self._getHashKey(name)
+        if imageKey in self.data:
+            del self.data[imageKey]
+            return True
+        else:
+            return False
+
+    def _getHashKey(self, imgName):
+        imageKey = hashlib.sha256(imgName.encode()).hexdigest()
+        return imageKey
+
+    def getCLIXML(self, imgName, cli):
+        imgKey = self._getHashKey(imgName)
+        if imgKey in self.data:
+            imageData = self.data[imgKey]
+
+        else:
+            print('no image named %s' % imgName)
+            return None
+
+        if cli in imageData[DockerCache.cli_dict]:
+
+            return imageData[DockerCache.cli_dict][cli][DockerCache.xml]
+        else:
+            print('no cli named %s' % imgName)
+            return None
+
+    def getCLIDict(self, imgName):
```

Comment:
@cdeepakroy 
I will refactor the name, but it is returning the json structure defined for each image which is why I though calling it a dict will be more appropriate.